File: typescript/base-hmi-types/src/screens/controls/HmiRecipeColumn.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiProperty } from "../base.js";
import { HmiRecipeColumnType } from "./HmiRecipeColumnType.js";

export class HmiRecipeColumn {
  type = HmiRecipeColumnType.Unknown;
  sourceType?: string;
  visible?: HmiProperty<boolean>;
  width?: HmiProperty<number>;
  headerText?: HmiMultilingualText;
}

File: src/BaseHmiTypes/Screens/Controls/HmiRecipeColumn.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiRecipeColumn
{
    public HmiRecipeColumnType Type { get; set; }

    public string? SourceType { get; set; }

    public HmiProperty<bool>? Visible { get; set; }

    public HmiProperty<double>? Width { get; set; }

    public HmiMultilingualText? HeaderText { get; set; }
}
